logger.gd

```diff
@@ -441,11 +494,19 @@ func get_default_output_level():
 # Output formatting
 # -----------------
 
-static func format(template, level, module, message):
+static func format(template, level, module, message, error_code = -1):
 	var output = template
 	output = output.replace(FORMAT_IDS.level, LEVELS[level])
 	output = output.replace(FORMAT_IDS.module, module)
 	output = output.replace(FORMAT_IDS.message, message)
+
+	# Error message substitution
+	var error_message = error_messages.get(error_code)
```

Comment:
Doesn't `get()` throw an error if the index is invalid?

Reply:
What's an example of an invalid index? Like passing an object or something? Otherwise, if the key doesn't exist, it returns `null` (or some default you specify) ([docs](https://docs.godotengine.org/en/stable/classes/class_dictionary.html#class-dictionary-method-get)).

Reply:
An invalid index would be `-1` (default value) or `404` (doesn't match any error code). But indeed, since `get()` defaults to returning `null` on failure, that's fine as is.